### PR TITLE
Add RL-10A4-2 & RL10A4-2N from BDB

### PR DESCRIPTION
**Pull request by vader111**
Add RL-10A4-2 & RL10A4-2N from BDB

### CODE REVIEW

**Comment by vader111** [comment]
Needs KSP-RO/RealismOverhaul#2377

**Comment by vader111** [comment]
(Ready To Review)

**Comment by vader111** [comment]
@Capkirk123 Should be Fixed now

File: GameData/ROEngines/PartConfigs/RL10A4-2N_BDB.cfg [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+	subcategory = 0
+	RSSROConfig = true
+	RP0conf = true
+
+	crashTolerance = 10
+	maxTemp = 673.15
+	skinTemp = 773.15
+	fuelCrossfeed = true
+	breakingForce = 10000
+	breakingTorque = 10000
```
Comment (DRVeyl) [inline GameData/ROEngines/PartConfigs/RL10A4-2N_BDB.cfg:17]: Make use of ROESetEngineDefaults

File: GameData/ROEngines/PartConfigs/RL10A4-2N_BDB.cfg [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+	tags = rl10, rl10a, rl10a-4-2n, 4, rl-10, hydrolox, centaur
+
+	engineType = RL10
+	stagingIcon = LIQUID_ENGINE
+	stageOffset = 1
+	childStageOffset = 1
+
+	TechRequired = hydrolox1992
+	cost = 500
+	entryCost = 26000
```
Comment (DRVeyl) [inline GameData/ROEngines/PartConfigs/RL10A4-2N_BDB.cfg:48]: Make use of ROESetEngineDefaults.  RP-1 must be responsible for where in the Tech Tree it ends up.

File: GameData/ROEngines/PartConfigs/RL10A4-2N_BDB.cfg [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+		engineID = RL10A4
+	}
+
+	MODULE
+	{
+		name = ModuleGimbal
+		gimbalTransformName = gimbalTransform
+		gimbalRange = 4
+		useGimbalResponseSpeed = true
+		gimbalResponseSpeed = 16
```
Comment (DRVeyl) [inline GameData/ROEngines/PartConfigs/RL10A4-2N_BDB.cfg:63]: Believe these values will be overridden by RO's engine config.  Should only set the module and transform in the original part def.  If the gimbal needs to be overrridden, that happens in a later pass.

File: GameData/ROEngines/PartConfigs/RL10A4-2N_BDB.cfg [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+		@configuration = RL10A-4-1-2
+		@origMass = 0.170
+		@CONFIG
+		{
+			@name = RL10A-4-1-2
+			@atmosphereCurve
+			{
+				@key = 0 451.0
+			}
+		}
```
Comment (DRVeyl) [inline GameData/ROEngines/PartConfigs/RL10A4-2N_BDB.cfg:196]: Which CONFIG entry are you editing?  There should be an RL10A-4-1-2 added in RO, and it should get the correct atmosphere curve, and this patch should just delete unwanted engine configs.  I guess this is sort of okay, but it's weird to use the engine config system and then edit it.  Unless there is a specific reason this config should not be in RO.

File: GameData/ROEngines/PartConfigs/RL10A4-2_BDB.cfg [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+	subcategory = 0
+	RSSROConfig = true
+	RP0conf = true
+
+	crashTolerance = 10
+	maxTemp = 673.15
+	skinTemp = 773.15
+	fuelCrossfeed = true
+	breakingForce = 10000
+	breakingTorque = 10000
```
Comment (DRVeyl) [inline GameData/ROEngines/PartConfigs/RL10A4-2_BDB.cfg:17]: Use ROESetEngineDefaults

File: GameData/ROEngines/PartConfigs/RL10A4-2_BDB.cfg [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+	tags = rl10, rl10a, rl10a-4-2, 4, rl-10, hydrolox, centaur
+
+	engineType = RL10
+	stagingIcon = LIQUID_ENGINE
+	stageOffset = 1
+	childStageOffset = 1
+
+	TechRequired = hydrolox1992
+	cost = 500
+	entryCost = 26000
```
Comment (DRVeyl) [inline GameData/ROEngines/PartConfigs/RL10A4-2_BDB.cfg:48]: Use ROESetEngineDefaults

File: GameData/ROEngines/PartConfigs/RL10A4-2_BDB.cfg [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+		engineID = RL10A4
+	}
+
+	MODULE
+	{
+		name = ModuleGimbal
+		gimbalTransformName = gimbalTransform
+		gimbalRange = 4
+		useGimbalResponseSpeed = true
+		gimbalResponseSpeed = 16
```
Comment (DRVeyl) [inline GameData/ROEngines/PartConfigs/RL10A4-2_BDB.cfg:63]: These are overwritten / configured by RO's engine config?

File: GameData/ROEngines/PartConfigs/RL10A4-2_BDB.cfg [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+	@MODULE[ModuleEngineConfigs] 
+	{
+		@configuration = RL10A-4-1-2
+		@origMass = 0.168
+		@CONFIG
+		{
+			@name = RL10A-4-1-2
+			@minThrust = 97.9
+			@maxThrust = 97.9
+		}
```
Comment (DRVeyl) [inline GameData/ROEngines/PartConfigs/RL10A4-2_BDB.cfg:200]: As above, seems this config should become present in RO.  Or why not?

**Review by DRVeyl — CHANGES_REQUESTED** [review]

**Comment by vader111** [comment]
Will need https://github.com/KSP-RO/RealismOverhaul/pull/2377

**Comment by vader111** [comment]
Also Needs https://github.com/KSP-RO/RP-0/pull/1406

**Review by DRVeyl — APPROVED** [review]

**Pull request merged**